### PR TITLE
Trans with master eclstate

### PR DESCRIPTION
**Pull request by blattms**
This incorporates the lastest changes to master from my PR OPM/opm-simulators#2243. It also needs
the changes to opm-grid. A backport of OPM/opm-grid#433 to the branch off time at Dec 10, 2019 can
be found in branch[noecl_base](https://github.com/blattms/opm-grid/tree/noecl_base).

Tested on SPE9. Not sure yet whether Norne works (there might be an issue if there is a GeoModifier
event as we might need to update more data).

### CODE REVIEW

File: ebos/eclcpgridvanguard.hh [hunk 1]
```diff
@@ -276,9 +278,30 @@ protected:
         // equilGrid_being a shallow copy only the global view.
         if (mpiRank == 0)
         {
+            const auto& gridProps = this->eclState(false).get3DProperties(); // this is hit
```
Comment (akva2) [inline ebos/eclcpgridvanguard.hh]: please remove the hit comment and let it assert.

File: ebos/eclcpgridvanguard.hh [hunk 2]
```diff
@@ -276,9 +278,30 @@ protected:
...
+            const std::vector<double>& porv = gridProps.getDoubleGridProperty("PORV").getData();
+            grid_->processEclipseFormat(&(this->eclState(true).getInputGrid()),
+                                    /*isPeriodic=*/false,
+                                    /*flipNormals=*/false,
+                                    /*clipZ=*/false,
+                                    porv,
+                                    this->eclState(false).getInputNNC()); // this is hit
```
Comment (akva2) [inline ebos/eclcpgridvanguard.hh]: ditto.

File: ebos/eclshallowgrid.hh [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+ * \author Dr. Blatt - HPC-Simulation-Software & Services
+ */
+
+#include <opm/parser/eclipse/EclipseState/Grid/EclipseGrid.hpp>
+#include <dune/common/parallel/collectivecommunication.hh>
+#include <dune/common/parallel/mpihelper.hh>
+#include <vector>
+namespace Opm
+{
+/// \brief Utility class that replicates cartesian grid data for prosity
```
Comment (akva2) [inline ebos/eclshallowgrid.hh]: porosity

Comment (blattms) [reply ebos/eclshallowgrid.hh]: done

File: ebos/eclcpgridvanguard.hh (hunk 2 above)
Comment (blattms) [reply ebos/eclcpgridvanguard.hh]: done

File: ebos/eclcpgridvanguard.hh (hunk 1 above)
Comment (blattms) [reply ebos/eclcpgridvanguard.hh]: done

**Comment by blattms** [comment]
Thanks. I amended the last commit.

File: ebos/eclcpgridvanguard.hh [hunk 4]
```diff
@@ -276,9 +278,30 @@ protected:
...
         if (mpiRank == 0)
         {
+            const auto& gridProps = this->eclState(true).get3DProperties();
+            const std::vector<double>& porv = gridProps.getDoubleGridProperty("PORV").getData();
+            grid_->processEclipseFormat(&(this->eclState(true).getInputGrid()),
+                                    /*isPeriodic=*/false,
+                                    /*flipNormals=*/false,
+                                    /*clipZ=*/false,
+                                    porv,
+                                    this->eclState(true).getInputNNC()); // this is hit
```
Comment (akva2) [inline ebos/eclcpgridvanguard.hh]: missed it first time. one more..

**Comment by blattms** [comment]
Norne is crashing.

**Comment by blattms** [comment]
or better: fails in an assertion
```
(gdb) up
#5  0x00005555573efca4 in Opm::EclOutputBlackOilModule<Opm::Properties::TTag::EclFlowProblem>::addRftDataToWells (this=0x555564a5d1d0, wellDatas=..., 
    reportStepNum=7)
    at /home/mblatt/src/dune/opm-2.6/opm-simulators/ebos/ecloutputblackoilmodule.hh:871
871                         const size_t index = simulator_.vanguard().eclState(true).getInputGrid().getGlobalIndex(i, j, k);
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff4be5535 in __GI_abort () at abort.c:79
#2  0x00007ffff4be540f in __assert_fail_base (fmt=0x7ffff4d47ee0 "%s%s%s:%u: %s%sAssertion `%s' failed.\n%n", assertion=0x555558dd5c99 "myRank == 0", 
    file=0x555558dcee58 "/home/mblatt/src/dune/opm-2.6/opm-simulators/ebos/eclbasevanguard.hh", line=398, function=<optimized out>) at assert.c:92
#3  0x00007ffff4bf3102 in __GI___assert_fail (assertion=0x555558dd5c99 "myRank == 0", 
    file=0x555558dcee58 "/home/mblatt/src/dune/opm-2.6/opm-simulators/ebos/eclbasevanguard.hh", line=398, 
    function=0x555558de58e0 <Opm::EclBaseVanguard<Opm::Properties::TTag::EclFlowProblem>::eclState(bool) const::__PRETTY_FUNCTION__> "const Opm::EclipseState& Opm::EclBaseVanguard<TypeTag>::eclState(bool) const [with TypeTag = Opm::Properties::TTag::EclFlowProblem]") at assert.c:101
#4  0x000055555739cf86 in Opm::EclBaseVanguard<Opm::Properties::TTag::EclFlowProblem>::eclState (this=0x555559edbc50, assert=true)
    at /home/mblatt/src/dune/opm-2.6/opm-simulators/ebos/eclbasevanguard.hh:398
#5  0x00005555573efca4 in Opm::EclOutputBlackOilModule<Opm::Properties::TTag::EclFlowProblem>::addRftDataToWells (this=0x555564a5d1d0, wellDatas=..., 
    reportStepNum=7) at /home/mblatt/src/dune/opm-2.6/opm-simulators/ebos/ecloutputblackoilmodule.hh:871
#6  0x00005555573b04cc in Opm::EclWriter<Opm::Properties::TTag::EclFlowProblem>::writeOutput (this=0x555564a5cff0, isSubStep=false)
    at /home/mblatt/src/dune/opm-2.6/opm-simulators/ebos/eclwriter.hh:362
#7  0x0000555557374d30 in Opm::EclProblem<Opm::Properties::TTag::EclFlowProblem>::writeOutput (this=0x555564a5bac0, verbose=true)
    at /home/mblatt/src/dune/opm-2.6/opm-simulators/ebos/eclproblem.hh:1094
#8  0x000055555735cf84 in Opm::SimulatorFullyImplicitBlackoilEbos<Opm::Properties::TTag::EclFlowProblem>::run (this=0x555559cf1a60, timer=...)
    at /home/mblatt/src/dune/opm-2.6/opm-simulators/opm/simulators/flow/SimulatorFullyImplicitBlackoilEbos.hpp:249
#9  0x0000555557344331 in Opm::FlowMainEbos<Opm::Properties::TTag::EclFlowProblem>::runSimulator (this=0x7fffffffa9f0, output_cout=false)
    at /home/mblatt/src/dune/opm-2.6/opm-simulators/opm/simulators/flow/FlowMainEbos.hpp:487
#10 0x000055555733628d in Opm::FlowMainEbos<Opm::Properties::TTag::EclFlowProblem>::execute (this=0x7fffffffa9f0, argc=2, argv=0x7fffffffd6e8, output_cout=false, 
    output_to_files=true) at /home/mblatt/src/dune/opm-2.6/opm-simulators/opm/simulators/flow/FlowMainEbos.hpp:261
#11 0x00005555572dcbf2 in Opm::flowEbosBlackoilMain (argc=2, argv=0x7fffffffd6e8, outputCout=false, outputFiles=true)
    at /home/mblatt/src/dune/opm-2.6/opm-simulators/flow/flow_ebos_blackoil.cpp:63
#12 0x00005555571ca1bd in main (argc=2, argv=0x7fffffffd6e8) at /home/mblatt/src/dune/opm-2.6/opm-simulators/flow/flow.cpp:463

```
on non-zero ranks during output.

Known issue?

**Comment by akva2** [comment]
no. probably due to updates in mainline that i have not addressed yet.

**Comment by blattms** [comment]
I must have introduced a subtle bug that breaks convergence for norne.

**Pull request closed without merging**